ckanext/switzerland/dcat/profiles.py

```diff
@@ -427,136 +405,122 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):  # noqa
             references = dataset_dict.get("see_alsos")
             for reference in references:
                 reference_identifier = reference["dataset_identifier"]
-                g.add(
-                    (dataset_ref, RDFS.seeAlso, Literal(reference_identifier))
-                )  # noqa
+                g.add((dataset_ref, RDFS.seeAlso, Literal(reference_identifier)))
 
         # Contact details
         if dataset_dict.get("contact_points"):
-            contact_points = self._get_dataset_value(
-                dataset_dict, "contact_points"
-            )  # noqa
+            contact_points = self._get_dataset_value(dataset_dict, "contact_points")
             for contact_point in contact_points:
                 contact_details = BNode()
                 contact_point_email = contact_point["email"]
                 contact_point_name = contact_point["name"]
 
                 g.add((contact_details, RDF.type, VCARD.Organization))
-                g.add(
-                    (contact_details, VCARD.hasEmail, URIRef(contact_point_email))
-                )  # noqa
+                g.add((contact_details, VCARD.hasEmail, URIRef(contact_point_email)))
                 g.add((contact_details, VCARD.fn, Literal(contact_point_name)))
 
                 g.add((dataset_ref, DCAT.contactPoint, contact_details))
 
         # Publisher
+        self._add_publishers_to_graph(dataset_dict, dataset_ref)
+
+        # Temporals
+        self._add_temporals_to_graph(dataset_dict, dataset_ref)
+
+        # Themes
+        g.add((dataset_ref, DCAT.theme, URIRef("http://opendata.swiss/group/mobility")))
+
+        # Resources
+        for resource_dict in dataset_dict.get("resources", []):
+            self._add_distribution_to_graph(dataset_ref, resource_dict)
+
+    def _add_publishers_to_graph(self, dataset_dict, dataset_ref):
         if dataset_dict.get("publishers"):
             publishers = dataset_dict.get("publishers")
             for publisher in publishers:
                 publisher_name = publisher["label"]
 
                 publisher_details = BNode()
-                g.add((publisher_details, RDF.type, DCT.Description))
-                g.add((publisher_details, RDFS.label, Literal(publisher_name)))
-                g.add((dataset_ref, DCT.publisher, publisher_details))
+                self.g.add((publisher_details, RDF.type, DCT.Description))
```

Comment:
here you could use as an example ogd script, also do implement multilingual names:
https://github.com/opendata-swiss/ckanext-dcatapchharvest/blob/e007878e6dde446cfa092594ce9eb000ecee3a2a/ckanext/dcatapchharvest/profiles.py#L1254